partiql-ast/src/main/java/org/partiql/ast/ddl/AttributeConstraint.java

```diff
@@ -0,0 +1,109 @@
+package org.partiql.ast.ddl;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.partiql.ast.AstNode;
+import org.partiql.ast.AstVisitor;
+import org.partiql.ast.expr.Expr;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * TODO docs, equals, hashcode
+ */
+public abstract class AttributeConstraint extends AstNode {
```

Comment:
Named attribute constraint because we use the same object in Struct Field construction. i.e., `Struct<foo: INT2 NOT NULL>`. 

The mental model is to have the grammar file follows SQL as close as possible, but rename certain things in AST to better fits our use case.

Reply:
From offline discussion, column here can be confusing since these constraints can be applied to structs and not just top-level table columns. Field is another consideration but applies to row types from the SQL1999 spec. And attribute refers to structured types. So leaving it as attribute seems reasonable so long as we add the appropriate docs in the future.